Website/DesktopModules/Hotcakes/Core/Controllers/CheckoutController.cs

```diff
@@ -104,7 +105,18 @@ public ActionResult Index()
             VerifySessionError(model);
             RenderErrorSummary(model);
             CheckFreeItems(model);
-
+            if (HccApp.CurrentStore.Settings.PaymentCreditCardGateway == "15011DF5-13DA-42BE-9DFF-31C71ED64D4A")
+            {
```

Comment:
There are a lot of values here without any defensive coding. Think about your testing... Do you think that's okay? If you think yes, that's fine. It's just a question. :)